hub/core/chunk_engine.py

```diff
@@ -459,12 +450,13 @@ def update(
     ):
         """Update data at `index` with `samples`."""
         self._write_initialization()
+        initial_autoflush = self.cache.autoflush
+        self.cache.autoflush = False
+
         if operator is not None:
             return self._update_with_operator(index, samples, operator)
 
         enc = self.chunk_id_encoder
-        updated_chunks = set()
```

Comment:
instead of maintaining a set of updated chunks and adding them to cache at the end, each chunk is always added to the cache dirty keys before it is modified (L472)